x-pack/plugins/apm/public/components/app/correlations/ml_failure_correlations.tsx

```diff
@@ -0,0 +1,410 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect, useMemo, useState } from 'react';
+import {
+  EuiCallOut,
+  EuiCode,
+  EuiAccordion,
+  EuiPanel,
+  EuiBasicTableColumn,
+  EuiButton,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiProgress,
+  EuiSpacer,
+  EuiText,
+  EuiBadge,
+  EuiIcon,
+  EuiLink,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { useHistory } from 'react-router-dom';
+import { useUrlParams } from '../../../context/url_params_context/use_url_params';
+import { useApmPluginContext } from '../../../context/apm_plugin/use_apm_plugin_context';
+import { CorrelationsTable } from './correlations_table';
+import { enableInspectEsQueries } from '../../../../../observability/public';
+import { asPreciseDecimal } from '../../../../common/utils/formatters';
+import { useApmServiceContext } from '../../../context/apm_service/use_apm_service_context';
+import { FailureCorrelationsHelpPopover } from './ml_failure_correlations_help_popover';
+import { useCorrelationsSearchStrategy } from './use_correlations_search_strategy';
+import {
+  BaseSearchStrategyResponse,
+  FailureCorrelationValue,
+} from '../../../../common/search_strategies/failure_correlations/types';
+import { ImpactBar } from '../../shared/ImpactBar';
+import { FAILURE_CORRELATION_SEARCH_STRATEGY } from '../../../../common/search_strategies/failure_correlations/constants';
+import { isErrorMessage } from './utils/is_error_message';
+import { Summary } from '../../shared/Summary';
+import { FETCH_STATUS } from '../../../hooks/use_fetcher';
+import { getFailureCorrelationImpactLabel } from './utils/get_failure_correlation_impact_label';
+import { createHref, push } from '../../shared/Links/url_helpers';
+import { useUiTracker } from '../../../../../observability/public';
+
+interface Props {
+  onClose: () => void;
+}
+
+interface ErrorCorrelationSearchStrategyResult
+  extends BaseSearchStrategyResponse {
+  values: FailureCorrelationValue[];
+}
+
+export function MlFailureCorrelations({ onClose }: Props) {
+  const {
+    core: { notifications, uiSettings },
+  } = useApmPluginContext();
+  const trackApmEvent = useUiTracker({ app: 'apm' });
+
+  const { serviceName, transactionType } = useApmServiceContext();
+  const { urlParams } = useUrlParams();
+
+  const { environment, kuery, transactionName, start, end } = urlParams;
+
+  const displayLog = uiSettings.get<boolean>(enableInspectEsQueries);
+
+  const result = useCorrelationsSearchStrategy<ErrorCorrelationSearchStrategyResult>(
+    {
+      ...{
+        ...{
+          environment,
+          kuery,
+          serviceName,
+          transactionName,
+          transactionType,
+          start,
+          end,
+        },
+      },
+    },
```

Comment:
Looks like this is unnecessarly deeply nested. I think it could just be the most inner object.

Reply:
Updated here https://github.com/elastic/kibana/pull/108441/commits/155443605ee491b2eb7c2bcb7b3e458eed9591e6